### ISSUE TITLE
Privacy plugin can't handle a large number of external assets

### ISSUE THREAD

**Issue by jonaharagon** [squidfunk/mkdocs-material#5127]
### Context

https://github.com/privacyguides/privacyguides.org/issues/2063#issuecomment-1451006044

### Bug description

On pages with a lot of external resources or links, the privacy plugin does not download all external assets. This is easily reproduced by just repeating the same emoji a bunch of times:

![image](https://user-images.githubusercontent.com/3637842/222289084-abd45f24-4869-482b-89c7-441b405fb884.png)

This is not only dependent on the number of external assets, any regular links to external sites also break external assets that come after them in the document:

![Screenshot 2023-03-01 at 17 21 58@2x](https://user-images.githubusercontent.com/3637842/222289419-5e5caa6e-77d8-4a93-afc9-5f9e9a204630.png)

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)

### Reproduction

[Archive.zip](https://github.com/squidfunk/mkdocs-material/files/10866109/Archive.zip)

### Steps to reproduce

1. Enable the privacy plugin
2. Load a large number of external images/assets (can be identical)

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and the excellent reproduction. Fixed in [`7e19ae71f`](https://github.com/squidfunk/mkdocs-material-insiders/commit/7e19ae71f0a872631b983ca7faa85e3c5d9fc981). In fact, it was no concurrency issues as initially suspected. The problem was that the flags passed to `re.sub` where interpreted as a maximum number of replacements, not as flags, since `count` is the first parameter in order. This lead to a maximum number of 40 replacements, as I counted.
